src/shared/lib/debounce.ts

```diff
@@ -0,0 +1,38 @@
+export default function debounce<AccumulatorType, ArgsType>(fn:(acc:AccumulatorType)=>void,
```

Comment:
I libraritized this to DRY it up, which I do think is necessary.

debounceAccumulate maybe?

Is this not a common pattern? debounce is a function creator, which includes its necessary variables in the closure of the returned function. I've seen and used this pattern before.